### DISCUSSION TITLE
Nesting calc()

### DISCUSSION THREAD

**Discussion by Fygo** [clientIO/joint#2007] [Q&A] [answered]
Hey guys,

So... 'calc()'. How do I use it properly? I can use it for simple expressions but anything a bit more nested/complicated and I can't get it to work.

Let's assume I need to count this:
w - 10 - h / 2

What would be the calc expression?
```
calc(calc(w - 10) - calc(h / 2)) // Uncaught Error: Invalid calc() expression: 210 - 30
calc(-calc(h) / 2 + calc(w - 10)) // Uncaught Error: Invalid calc() expression: -60 / 2 + 210
calc(w - 10 - calc(h / 2)) // Uncaught Error: Invalid calc() expression: w - 10 - 30
```
...etc.

This expression cannot be counted using calc?
Thank you.

**Comment by kumilingus** [comment, marked as answer]
It can.
```js
calc(w - calc(h / 2 + 10))
```

**Comment by Fygo** [comment]
Thanks!
So I need to strictly adhere to those patterns/expressions you have listed in the docs?

Just out of curiosity - why didn't (e.g.) the first one work? _calc(calc(w - 10) - calc(h / 2))_

**Reply by kumilingus** [reply]
You can try to evaluate the nested expression `calc` first and then substitute the parent expression with their values. It must remain a valid expression.

e.g. for `width=100` and `height=80`

```js
calc(w - calc(h / 2 + 10)) // calc(h / 2 + 10) => 80 / 2 + 10 = 50
calc(w - 50); // 100 - 50
50;
```

while

```js
calc(calc(w - 10) - calc(h / 2)); // calc(w - 10) => 100 - 10 = 90
calc(90 - calc(h / 2)); // Error: Is not a valid expression, `calc` it expects a variable first
```

Note that `calc` implements these strict rules to achieve better parsing performance.
